tools/pythonpkg/src/pyresult.cpp

```diff
@@ -217,8 +217,14 @@ py::dict DuckDBPyResult::FetchNumpyInternal(bool stream, idx_t vectors_per_chunk
 
 	// now that we have materialized the result in contiguous arrays, construct the actual NumPy arrays
 	py::dict res;
+	unordered_map<string, idx_t> names;
 	for (idx_t col_idx = 0; col_idx < result->types.size(); col_idx++) {
-		res[result->names[col_idx].c_str()] = conversion.ToArray(col_idx);
+		if (names[result->names[col_idx]]++ == 0) {
+			res[result->names[col_idx].c_str()] = conversion.ToArray(col_idx);
+		} else {
+			res[(result->names[col_idx] + "_" + to_string(names[result->names[col_idx]])).c_str()] =
```

Comment:
What if there is a naming conflict here? Perhaps detect this and add another number (e.g. `_1`, `_2`, etc)?

Reply:
I don't exactly know what you mean, can you send me a concrete example?

Reply:
Suppose I have three columns, "id", "id" and "id_1".

I will now rename my second column "id" to "id_1", what happens to the column that was originally called "id_1"?

Reply:
There is no particular reason for using an underscore, since this is related to pandas, maybe there is a preferred symbol for pandas?

Reply:
Pandas allows duplicate column names (Gah!) so I'm not sure if there is a standard there that can be followed.

Reply:
@Mytherin any opinions? I'm not married with the underscore, so I'm fine with something else.

Reply:
Underscore or colon are both fine by me. Does Pandas do anything there itself?